Refit/FormValueDictionary.cs

```diff
@@ -45,15 +45,74 @@ public FormValueDictionary(object source, RefitSettings settings)
                     var value = property.GetValue(source, null);
                     if (value != null)
                     {
+                        var fieldName = GetFieldNameForProperty(property);
+
                         // see if there's a query attribute
                         var attrib = property.GetCustomAttribute<QueryAttribute>(true);
 
-                        Add(GetFieldNameForProperty(property), settings.FormUrlEncodedParameterFormatter.Format(value, attrib?.Format));
+                        if (value is IEnumerable enumerable) {
+                            switch (attrib?.CollectionFormat) {
+                                case CollectionFormat.Multi:
+                                    foreach (var item in enumerable) {
+                                        Add(fieldName, settings.FormUrlEncodedParameterFormatter.Format(item, null));
+                                    }
+                                    break;
+                                case CollectionFormat.Csv:
+                                case CollectionFormat.Ssv:
+                                case CollectionFormat.Tsv:
+                                case CollectionFormat.Pipes:
+                                    var delimiter = attrib.CollectionFormat == CollectionFormat.Csv ?  ","
+                                        : attrib.CollectionFormat == CollectionFormat.Ssv ? " "
+                                        : attrib.CollectionFormat == CollectionFormat.Tsv ? "\t" : "|";
+
+                                    var formattedValues = enumerable
+                                        .Cast<object>()
+                                        .Select(v => settings.FormUrlEncodedParameterFormatter.Format(v, null));
```

Comment:
Is there still value providing the attribute format string? Could these be combined? For example, for a specific double formatting in addition to the collection format?

Reply:
Wasn't sure before how the format strings fit into the picture. The attribute format string should _definitely_ be respected